python/ml4ir/base/config/keys.py

```diff
@@ -137,6 +137,9 @@ class MonteCarloInferenceKey(Key):
     MONTE_CARLO_TRIALS = "monte_carlo_trials"
     NUM_TEST_TRIALS = "num_test_trials"
     NUM_TRAINING_TRIALS = "num_training_trials"
+    USE_FIXED_MASK_IN_TRAINING = "use_fixed_mask_in_training"
+    USE_FIXED_MASK_IN_TESTING = "use_fixed_mask_in_testing"
+    FIXED_MASK_COUNT = "fixed_mask_count"
```

Comment:
@mohazahran Isn't `FIXED_MASK_COUNT` the same as `NUM_TRAINING_TRIALS`? Why do we need two args? Also, why is there no `FIXED_MASK_COUNT_TRAIN` and `FIXED_MASK_COUNT_TEST`?

Reply:
No they are different. I will add more comments to clear this confusion. I'm adding the support of having both stochastic and deterministic MC for training and testing. So you can choose for example to have stochastic MC in training by setting `NUM_TRAINING_TRIALS` and have deterministic MC in testing by setting `USE_FIXED_MASK_IN_TESTING`